### PR TITLE
GA AAD

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java [hunk 1]
```diff
@@ -345,7 +347,6 @@ public static class PreviewConfiguration {
 
     public ProfilerConfiguration profiler = new ProfilerConfiguration();
     public GcEventConfiguration gcEvents = new GcEventConfiguration();
-    public AadAuthentication authentication = new AadAuthentication();
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java:348]: can we keep here and copy over in `ConfigurationBuilder` and log warning? e.g. something like

https://github.com/microsoft/ApplicationInsights-Java/blob/383ccc9a4cb2ee03e693113c677d152cd784d2ac/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java#L221-L232

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java:348]: we have AadAuthentication.validate() that takes care of it

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java [hunk 2]
```diff
@@ -62,6 +63,7 @@ private static boolean isEmpty(@Nullable String str) {
 
   public void validate() {
     instrumentation.logging.getSeverityThreshold();
+    authentication.validate();
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java:66]: here is the validation

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 3]
```diff
@@ -230,6 +230,11 @@ private static void logConfigurationWarnings(Configuration config) {
         config.preview.connectionStringOverrides.add(newOverride);
       }
     }
+    if (config.preview.authentication.enabled) {
+      configurationLogger.warn(
+          "\"authentication\" is no longer in preview and it has been GA since 3.4.18");
+      config.authentication.enabled = true;
+    }
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:237]: can we copy over the other aad config values here too?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 4]
```diff
@@ -230,6 +230,11 @@ private static void logConfigurationWarnings(Configuration config) {
         config.preview.connectionStringOverrides.add(newOverride);
       }
     }
+    if (config.preview.authentication.enabled) {
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:233]:
```suggestion
    if (config.preview.authentication.enabled && !config.authentication.enabled) {
```

**Review by trask — APPROVED** [review]

**Comment by trask** [comment]
@heyams can you queue up a doc PR also for this?

**Comment by heyams** [comment]
> @heyams can you queue up a doc PR also for this?

https://github.com/MicrosoftDocs/azure-docs-pr/pull/251370

**Pull request merged**